execution-engine/src/hcall/wasmi.rs

```diff
@@ -529,30 +513,30 @@ impl WasmiHostProvisioningState {
         let address: u32 = args.nth(1);
 
         match self.get_memory() {
-            None => Err(FatalHostError::NoMemoryRegistered),
-            Some(memory) => match self.get_current_data_source(index as usize) {
-                None => return Ok(VeracruzError::BadInput),
-                Some(frame) => {
-                    let result = frame.get_data().len() as u32;
-                    let result: Vec<u8> = result.to_le_bytes().to_vec();
-
-                    if let Err(_) = memory.set(address, &result) {
-                        return Err(FatalHostError::MemoryWriteFailed {
-                            memory_address: address as usize,
-                            bytes_to_be_written: result.len() as usize,
-                        });
-                    }
+            None => Err(FatalEngineError::NoMemoryRegistered),
+            Some(memory) => 
+            {
+                //TODO FILL IN principal
+                let result = self.read_file(&self.program,&format!("input-{}",index))?.ok_or(format!("File input-{} cannot be found",index))?.len() as u32;
+                let result: Vec<u8> = result.to_le_bytes().to_vec();
 
-                    Ok(VeracruzError::Success)
+                if let Err(_) = memory.set(address, &result) {
```

Comment:
This is probably neater to write:

```rust
if memory.set(address, &result).is_err() {
  ...
}
```

as the actual error is not being bound in the pattern matching.